dev-util/xmake/patches/xmake-2.8.5.patchset

```diff
@@ -0,0 +1,54 @@
+From fbabef9f25bcfb805ebd8a68482554957001cfa8 Mon Sep 17 00:00:00 2001
+From: Christof Meerwald <cmeerw@cmeerw.org>
+Date: Fri, 15 Dec 2023 14:56:01 +0000
+Subject: [PATCH 1/2] adjust fallback for Haiku OS
+
+---
+ core/src/xmake/engine.c | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/core/src/xmake/engine.c b/core/src/xmake/engine.c
+index f1b0b381b..a24058591 100644
+--- a/core/src/xmake/engine.c
++++ b/core/src/xmake/engine.c
+@@ -682,8 +682,12 @@ static tb_size_t xm_engine_get_program_file(xm_engine_t* engine, tb_char_t* path
+         static tb_char_t const* s_paths[] =
+         {
+             "~/.local/bin/xmake",
++#if defined(TB_CONFIG_OS_HAIKU)
++            "/Haiku/system/bin/xmake"
++#else
```

Comment:
This should be `/boot/system/...`.  The disk isn't guaranteed to be named `Haiku`.

A better option would be to use the `FindDirectory` or `PathFinder` API to search for `xmake` in any bin directory, not just `/system/bin`

Reply:
Also got a suggestion on discuss.haiku-os.org for other options here, but that's for another day

Reply:
Yeah, that was also me on the forums :stuck_out_tongue: 

I didn't look close enough at the code to see that it only needed the current exe path.  In that case `get_next_image_info()` is probably the way to go to do it properly.

Depending on what app you're porting you can sometimes get away with reckoning the path using `argv[0]`.  A lot of the SDL games do `chdir(dirname(argv[0]))` so they can find their data directory.